packages/sign-client/src/controllers/engine.ts

```diff
@@ -154,15 +165,14 @@ export class Engine extends IEngine {
   public approve: IEngine["approve"] = async (params) => {
     this.isInitialized();
     await this.isValidApprove(params);
-    const { id, relayProtocol, namespaces } = params;
+    const { id, relayProtocol, namespaces, sessionProperties } = params;
     const proposal = this.client.proposal.get(id);
-    let { pairingTopic, proposer, requiredNamespaces } = proposal;
-
+    let { pairingTopic, proposer, requiredNamespaces, optionalNamespaces } = proposal;
     if (!isValidObject(requiredNamespaces)) {
       const required = getRequiredNamespacesFromNamespaces(namespaces, "approve()");
       requiredNamespaces = required;
-      // update the proposal with the new required namespaces
-      this.client.proposal.set(id, { ...proposal, requiredNamespaces });
+      // update the proposal with the new required & optional namespaces
+      this.client.proposal.set(id, { ...proposal, requiredNamespaces, optionalNamespaces });
```

Comment:
@ganchoradkov isn't this missing `sessionProperties`?

Reply:
Yes indeed, and I see this could be optimized as just after 21 lines the proposal is deleted so the whole update is redundant